### PR TITLE
show output diff for test failures

### PR DESCRIPTION
**Pull request by SolidifiedRay**
Use Python's difflib to show the diff of the outputs for the demo tests.

ref: [issue#16](https://github.com/in-toto/demo/issues/16)

### CODE REVIEW

File: run_demo_md.py [hunk 1]
```diff
@@ -27,6 +27,8 @@
 import sys
 import tempfile
 import six
+import difflib
+from pprint import pprint
```
Comment (lukpueh) [inline run_demo_md.py:31]: No need to import pprint.

**Review by lukpueh — COMMENTED** [review]
Cool stuff, @SolidifiedRay! Could you show us a sample output?

**Comment by SolidifiedRay** [comment]
The following is an example of the ouput: 

```
#### EXPECTED:
-
+ cd owner_alice
+ python create_layout.py
+ cd ../functionary_bob
+ in-toto-run --step-name clone --products demo-project/foo.py --key bob -- git clone https://github.com/in-toto/demo-project.git
+ in-toto-record start --step-name update-version --key bob --materials demo-project/foo.py
+ cat
+ in-toto-record stop --step-name update-version --key bob --products demo-project/foo.py
+ cp -r demo-project ../functionary_carl/
+ cd ../functionary_carl
+ in-toto-run --step-name package --materials demo-project/foo.py --products demo-project.tar.gz --key carl -- tar --exclude .git -zcvf demo-project.tar.gz demo-project
+ cd ..
+ cp owner_alice/root.layout functionary_bob/clone.776a00e2.link functionary_bob/update-version.776a00e2.link functionary_carl/package.2f89b927.link functionary_carl/demo-project.tar.gz final_product/
+ cd final_product
+ cp ../owner_alice/alice.pub .
+ in-toto-verify --layout root.layout --layout-key alice.pub
+ echo 0
0
+ cd ../functionary_carl
+ echo something evil
+ in-toto-run --step-name package --materials demo-project/foo.py --products demo-project.tar.gz --key carl -- tar --exclude .git -zcvf demo-project.tar.gz demo-project
+ cd ..
+ cp owner_alice/root.layout functionary_bob/clone.776a00e2.link functionary_bob/update-version.776a00e2.link functionary_carl/package.2f89b927.link functionary_carl/demo-project.tar.gz final_product/
+ cd final_product
+ in-toto-verify --layout root.layout --layout-key alice.pub
(in-toto-verify) RuleVerificationError: 'DISALLOW *' matched the following artifacts: [u'demo-project/foo.py']
Full trace for 'expected_materials' of item 'package':
Available materials (used for queue):
[u'demo-project/foo.py']
Available products:
[u'demo-project.tar.gz']
Queue after 'MATCH demo-project/* WITH PRODUCTS FROM update-version':
[u'demo-project/foo.py']

+ echo 1
1

-
#### GOT:
-
+ cd owner_alice
+ python create_layout.py
/bin/sh: 5: python: not found
+ cd ../functionary_bob
+ in-toto-run --step-name clone --products demo-project/foo.py --key bob -- git clone https://github.com/in-toto/demo-project.git
+ in-toto-record start --step-name update-version --key bob --materials demo-project/foo.py
+ cat
+ in-toto-record stop --step-name update-version --key bob --products demo-project/foo.py
+ cp -r demo-project ../functionary_carl/
+ cd ../functionary_carl
+ in-toto-run --step-name package --materials demo-project/foo.py --products demo-project.tar.gz --key carl -- tar --exclude .git -zcvf demo-project.tar.gz demo-project
+ cd ..
+ cp owner_alice/root.layout functionary_bob/clone.776a00e2.link functionary_bob/update-version.776a00e2.link functionary_carl/package.2f89b927.link functionary_carl/demo-project.tar.gz final_product/
cp: cannot stat 'owner_alice/root.layout': No such file or directory
+ cd final_product
+ cp ../owner_alice/alice.pub .
+ in-toto-verify --layout root.layout --layout-key alice.pub
(in-toto-verify) IOError: [Errno 2] No such file or directory: 'root.layout'
+ echo 1
1
+ cd ../functionary_carl
+ echo something evil
+ in-toto-run --step-name package --materials demo-project/foo.py --products demo-project.tar.gz --key carl -- tar --exclude .git -zcvf demo-project.tar.gz demo-project
+ cd ..
+ cp owner_alice/root.layout functionary_bob/clone.776a00e2.link functionary_bob/update-version.776a00e2.link functionary_carl/package.2f89b927.link functionary_carl/demo-project.tar.gz final_product/
cp: cannot stat 'owner_alice/root.layout': No such file or directory
+ cd final_product
+ in-toto-verify --layout root.layout --layout-key alice.pub
(in-toto-verify) IOError: [Errno 2] No such file or directory: 'root.layout'
+ echo 1
1

-
#### DIFFERENCE:
-
  + cd owner_alice
  + python create_layout.py
+ /bin/sh: 5: python: not found
  + cd ../functionary_bob
  + in-toto-run --step-name clone --products demo-project/foo.py --key bob -- git clone https://github.com/in-toto/demo-project.git
  + in-toto-record start --step-name update-version --key bob --materials demo-project/foo.py
  + cat
  + in-toto-record stop --step-name update-version --key bob --products demo-project/foo.py
  + cp -r demo-project ../functionary_carl/
  + cd ../functionary_carl
  + in-toto-run --step-name package --materials demo-project/foo.py --products demo-project.tar.gz --key carl -- tar --exclude .git -zcvf demo-project.tar.gz demo-project
  + cd ..
  + cp owner_alice/root.layout functionary_bob/clone.776a00e2.link functionary_bob/update-version.776a00e2.link functionary_carl/package.2f89b927.link functionary_carl/demo-project.tar.gz final_product/
+ cp: cannot stat 'owner_alice/root.layout': No such file or directory
  + cd final_product
  + cp ../owner_alice/alice.pub .
  + in-toto-verify --layout root.layout --layout-key alice.pub
+ (in-toto-verify) IOError: [Errno 2] No such file or directory: 'root.layout'
- + echo 0
?        ^

+ + echo 1
?        ^

- 0
+ 1
  + cd ../functionary_carl
  + echo something evil
  + in-toto-run --step-name package --materials demo-project/foo.py --products demo-project.tar.gz --key carl -- tar --exclude .git -zcvf demo-project.tar.gz demo-project
  + cd ..
  + cp owner_alice/root.layout functionary_bob/clone.776a00e2.link functionary_bob/update-version.776a00e2.link functionary_carl/package.2f89b927.link functionary_carl/demo-project.tar.gz final_product/
+ cp: cannot stat 'owner_alice/root.layout': No such file or directory
  + cd final_product
  + in-toto-verify --layout root.layout --layout-key alice.pub
+ (in-toto-verify) IOError: [Errno 2] No such file or directory: 'root.layout'
- (in-toto-verify) RuleVerificationError: 'DISALLOW *' matched the following artifacts: [u'demo-project/foo.py']
- Full trace for 'expected_materials' of item 'package':
- Available materials (used for queue):
- [u'demo-project/foo.py']
- Available products:
- [u'demo-project.tar.gz']
- Queue after 'MATCH demo-project/* WITH PRODUCTS FROM update-version':
- [u'demo-project/foo.py']
- 
  + echo 1
  1
-
Demo test failed due to unexpected output (see above). :(

```

**Comment by lukpueh** [comment]
Yeah! Exactly what I wanted. :) I think you can remove the old "EXPECTED" and "GOT" output. Also, can you please remove the hyphens around the output placeholder, i.e.  `-\n{}\n-`  becomes `\n{}\n`? I had put them there to mark begin and end of the output, but now with the diff view that's not necessary anymore, on the contrary, I think it adds confusion.

**Comment by SolidifiedRay** [comment]
> Also, can you please remove the hyphens around the output placeholder.
Sure! I have updated the changes.

**Review by lukpueh — APPROVED** [review]
🎉  Solid work, @SolidifiedRay! Thanks.

**Pull request merged**